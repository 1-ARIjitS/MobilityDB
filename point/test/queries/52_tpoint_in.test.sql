-----------------------------------------------------------------------

SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint 'Point(1 2)@2000-01-01')));
SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint '{Point(1 2)@2000-01-01, Point(3 4)@2000-01-02}')));
SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint '[Point(1 2)@2000-01-01, Point(3 4)@2000-01-02]')));
SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint '[Point(1 2)@2000-01-01, Point(3 4)@2000-01-02)')));
SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint '(Point(1 2)@2000-01-01, Point(3 4)@2000-01-02]')));
SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint '(Point(1 2)@2000-01-01, Point(3 4)@2000-01-02)')));
SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint '{[Point(1 2)@2000-01-01, Point(3 4)@2000-01-02],[Point(1 2)@2000-01-03, Point(3 4)@2000-01-04]}')));

SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint 'SRID=4326;Point(1 2 3)@2000-01-01',1,2)));
SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint 'SRID=4326;{Point(1 2 3)@2000-01-01, Point(4 5 6)@2000-01-02}',1,2)));
SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint 'SRID=4326;[Point(1 2 3)@2000-01-01, Point(4 5 6)@2000-01-02]',1,2)));
SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint 'SRID=4326;{[Point(1 2 3)@2000-01-01, Point(4 5 6)@2000-01-02],[Point(1 2 3)@2000-01-03, Point(4 5 6)@2000-01-04]}',1,2)));
/* Errors */
SELECT asEWKT(fromMFJSON('{"type":"MovingPoint","coordinates":[1],"datetimes":"2000-01-01T00:00:00+01","interpolations":["Discrete"]}'));
SELECT asEWKT(fromMFJSON('{"type":"MovingPoint","coordinates":[1,2,3,4],"datetimes":"2000-01-01T00:00:00+01","interpolations":["Discrete"]}'));
SELECT asEWKT(fromMFJSON('{"type":"MovingPoint","coordinatess":[1],"datetimes":"2000-01-01T00:00:00+01","interpolations":["Discrete"]}'));
SELECT asEWKT(fromMFJSON('{"type":"MovingPoint","coordinates":[1],"datetimess":"2000-01-01T00:00:00+01","interpolations":["Discrete"]}'));
SELECT asEWKT(fromMFJSON('{"type":"MovingPoint","coordinates":[1,2],"datetimess":"2000-01-01T00:00:00+01","interpolations":["Discrete"]}'));
SELECT asEWKT(fromMFJSON('{"type":"MovingPoint","coordinates":[[1,2],[1,2]],"datetimes":["2000-01-01T00:00:00+01"],"lower_inc":true,"upper_inc":true,"interpolations":["Linear"]}'));

-----------------------------------------------------------------------

SELECT asEWKT(fromEWKB(asEWKB(tgeompoint 'Point(1 2)@2000-01-01')));
SELECT asEWKT(fromEWKB(asEWKB(tgeompoint '{Point(1 2)@2000-01-01, Point(3 4)@2000-01-02}')));
SELECT asEWKT(fromEWKB(asEWKB(tgeompoint '[Point(1 2)@2000-01-01, Point(3 4)@2000-01-02]')));
SELECT asEWKT(fromEWKB(asEWKB(tgeompoint '[Point(1 2)@2000-01-01, Point(3 4)@2000-01-02)')));
SELECT asEWKT(fromEWKB(asEWKB(tgeompoint '(Point(1 2)@2000-01-01, Point(3 4)@2000-01-02]')));
SELECT asEWKT(fromEWKB(asEWKB(tgeompoint '(Point(1 2)@2000-01-01, Point(3 4)@2000-01-02)')));
SELECT asEWKT(fromEWKB(asEWKB(tgeompoint '{[Point(1 2)@2000-01-01, Point(3 4)@2000-01-02],[Point(1 2)@2000-01-03, Point(3 4)@2000-01-04]}')));

SELECT asEWKT(fromEWKB(asEWKB(tgeompoint 'SRID=4326;Point(1 2 3)@2000-01-01')));
SELECT asEWKT(fromEWKB(asEWKB(tgeompoint 'SRID=4326;{Point(1 2 3)@2000-01-01, Point(4 5 6)@2000-01-02}')));
SELECT asEWKT(fromEWKB(asEWKB(tgeompoint 'SRID=4326;[Point(1 2 3)@2000-01-01, Point(4 5 6)@2000-01-02]')));
SELECT asEWKT(fromEWKB(asEWKB(tgeompoint 'SRID=4326;{[Point(1 2 3)@2000-01-01, Point(4 5 6)@2000-01-02],[Point(1 2 3)@2000-01-03, Point(4 5 6)@2000-01-04]}')));

-----------------------------------------------------------------------
