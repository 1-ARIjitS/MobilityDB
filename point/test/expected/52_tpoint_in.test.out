SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint 'Point(1 2)@2000-01-01')));
              asewkt               
-----------------------------------
 POINT(1 2)@2000-01-01 00:00:00+00
(1 row)

SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint '{Point(1 2)@2000-01-01, Point(3 4)@2000-01-02}')));
                                 asewkt                                 
------------------------------------------------------------------------
 {POINT(1 2)@2000-01-01 00:00:00+00, POINT(3 4)@2000-01-02 00:00:00+00}
(1 row)

SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint '[Point(1 2)@2000-01-01, Point(3 4)@2000-01-02]')));
                                 asewkt                                 
------------------------------------------------------------------------
 [POINT(1 2)@2000-01-01 00:00:00+00, POINT(3 4)@2000-01-02 00:00:00+00]
(1 row)

SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint '[Point(1 2)@2000-01-01, Point(3 4)@2000-01-02)')));
                                 asewkt                                 
------------------------------------------------------------------------
 [POINT(1 2)@2000-01-01 00:00:00+00, POINT(3 4)@2000-01-02 00:00:00+00)
(1 row)

SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint '(Point(1 2)@2000-01-01, Point(3 4)@2000-01-02]')));
                                 asewkt                                 
------------------------------------------------------------------------
 (POINT(1 2)@2000-01-01 00:00:00+00, POINT(3 4)@2000-01-02 00:00:00+00]
(1 row)

SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint '(Point(1 2)@2000-01-01, Point(3 4)@2000-01-02)')));
                                 asewkt                                 
------------------------------------------------------------------------
 (POINT(1 2)@2000-01-01 00:00:00+00, POINT(3 4)@2000-01-02 00:00:00+00)
(1 row)

SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint '{[Point(1 2)@2000-01-01, Point(3 4)@2000-01-02],[Point(1 2)@2000-01-03, Point(3 4)@2000-01-04]}')));
                                                                      asewkt                                                                      
--------------------------------------------------------------------------------------------------------------------------------------------------
 {[POINT(1 2)@2000-01-01 00:00:00+00, POINT(3 4)@2000-01-02 00:00:00+00], [POINT(1 2)@2000-01-03 00:00:00+00, POINT(3 4)@2000-01-04 00:00:00+00]}
(1 row)

SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint 'SRID=4326;Point(1 2 3)@2000-01-01',1,2)));
                      asewkt                      
--------------------------------------------------
 SRID=4326;POINT Z (1 2 3)@2000-01-01 00:00:00+00
(1 row)

SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint 'SRID=4326;{Point(1 2 3)@2000-01-01, Point(4 5 6)@2000-01-02}',1,2)));
                                           asewkt                                           
--------------------------------------------------------------------------------------------
 SRID=4326;{POINT Z (1 2 3)@2000-01-01 00:00:00+00, POINT Z (4 5 6)@2000-01-02 00:00:00+00}
(1 row)

SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint 'SRID=4326;[Point(1 2 3)@2000-01-01, Point(4 5 6)@2000-01-02]',1,2)));
                                           asewkt                                           
--------------------------------------------------------------------------------------------
 SRID=4326;[POINT Z (1 2 3)@2000-01-01 00:00:00+00, POINT Z (4 5 6)@2000-01-02 00:00:00+00]
(1 row)

SELECT asEWKT(fromMFJSON(asMFJSON(tgeompoint 'SRID=4326;{[Point(1 2 3)@2000-01-01, Point(4 5 6)@2000-01-02],[Point(1 2 3)@2000-01-03, Point(4 5 6)@2000-01-04]}',1,2)));
                                                                                     asewkt                                                                                     
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 SRID=4326;{[POINT Z (1 2 3)@2000-01-01 00:00:00+00, POINT Z (4 5 6)@2000-01-02 00:00:00+00], [POINT Z (1 2 3)@2000-01-03 00:00:00+00, POINT Z (4 5 6)@2000-01-04 00:00:00+00]}
(1 row)

/* Errors */
SELECT asEWKT(fromMFJSON('{"type":"MovingPoint","coordinates":[1],"datetimes":"2000-01-01T00:00:00+01","interpolations":["Discrete"]}'));
ERROR:  Too few coordinates in MFJSON string
SELECT asEWKT(fromMFJSON('{"type":"MovingPoint","coordinates":[1,2,3,4],"datetimes":"2000-01-01T00:00:00+01","interpolations":["Discrete"]}'));
ERROR:  Too many coordinates in MFJSON string
SELECT asEWKT(fromMFJSON('{"type":"MovingPoint","coordinatess":[1],"datetimes":"2000-01-01T00:00:00+01","interpolations":["Discrete"]}'));
ERROR:  Unable to find 'coordinates' in MFJSON string
SELECT asEWKT(fromMFJSON('{"type":"MovingPoint","coordinates":[1],"datetimess":"2000-01-01T00:00:00+01","interpolations":["Discrete"]}'));
ERROR:  Too few coordinates in MFJSON string
SELECT asEWKT(fromMFJSON('{"type":"MovingPoint","coordinates":[1,2],"datetimess":"2000-01-01T00:00:00+01","interpolations":["Discrete"]}'));
ERROR:  Unable to find 'datetimes' in MFJSON string
SELECT asEWKT(fromMFJSON('{"type":"MovingPoint","coordinates":[[1,2],[1,2]],"datetimes":["2000-01-01T00:00:00+01"],"lower_inc":true,"upper_inc":true,"interpolations":["Linear"]}'));
ERROR:  Distinct number of elements in 'coordinates' and 'datetimes' arrays
SELECT asEWKT(fromEWKB(asEWKB(tgeompoint 'Point(1 2)@2000-01-01')));
              asewkt               
-----------------------------------
 POINT(1 2)@2000-01-01 00:00:00+00
(1 row)

SELECT asEWKT(fromEWKB(asEWKB(tgeompoint '{Point(1 2)@2000-01-01, Point(3 4)@2000-01-02}')));
                                 asewkt                                 
------------------------------------------------------------------------
 {POINT(1 2)@2000-01-01 00:00:00+00, POINT(3 4)@2000-01-02 00:00:00+00}
(1 row)

SELECT asEWKT(fromEWKB(asEWKB(tgeompoint '[Point(1 2)@2000-01-01, Point(3 4)@2000-01-02]')));
                                 asewkt                                 
------------------------------------------------------------------------
 [POINT(1 2)@2000-01-01 00:00:00+00, POINT(3 4)@2000-01-02 00:00:00+00]
(1 row)

SELECT asEWKT(fromEWKB(asEWKB(tgeompoint '[Point(1 2)@2000-01-01, Point(3 4)@2000-01-02)')));
                                 asewkt                                 
------------------------------------------------------------------------
 [POINT(1 2)@2000-01-01 00:00:00+00, POINT(3 4)@2000-01-02 00:00:00+00)
(1 row)

SELECT asEWKT(fromEWKB(asEWKB(tgeompoint '(Point(1 2)@2000-01-01, Point(3 4)@2000-01-02]')));
                                 asewkt                                 
------------------------------------------------------------------------
 (POINT(1 2)@2000-01-01 00:00:00+00, POINT(3 4)@2000-01-02 00:00:00+00]
(1 row)

SELECT asEWKT(fromEWKB(asEWKB(tgeompoint '(Point(1 2)@2000-01-01, Point(3 4)@2000-01-02)')));
                                 asewkt                                 
------------------------------------------------------------------------
 (POINT(1 2)@2000-01-01 00:00:00+00, POINT(3 4)@2000-01-02 00:00:00+00)
(1 row)

SELECT asEWKT(fromEWKB(asEWKB(tgeompoint '{[Point(1 2)@2000-01-01, Point(3 4)@2000-01-02],[Point(1 2)@2000-01-03, Point(3 4)@2000-01-04]}')));
                                                                      asewkt                                                                      
--------------------------------------------------------------------------------------------------------------------------------------------------
 {[POINT(1 2)@2000-01-01 00:00:00+00, POINT(3 4)@2000-01-02 00:00:00+00], [POINT(1 2)@2000-01-03 00:00:00+00, POINT(3 4)@2000-01-04 00:00:00+00]}
(1 row)

SELECT asEWKT(fromEWKB(asEWKB(tgeompoint 'SRID=4326;Point(1 2 3)@2000-01-01')));
                      asewkt                      
--------------------------------------------------
 SRID=4326;POINT Z (1 2 3)@2000-01-01 00:00:00+00
(1 row)

SELECT asEWKT(fromEWKB(asEWKB(tgeompoint 'SRID=4326;{Point(1 2 3)@2000-01-01, Point(4 5 6)@2000-01-02}')));
                                           asewkt                                           
--------------------------------------------------------------------------------------------
 SRID=4326;{POINT Z (1 2 3)@2000-01-01 00:00:00+00, POINT Z (4 5 6)@2000-01-02 00:00:00+00}
(1 row)

SELECT asEWKT(fromEWKB(asEWKB(tgeompoint 'SRID=4326;[Point(1 2 3)@2000-01-01, Point(4 5 6)@2000-01-02]')));
                                           asewkt                                           
--------------------------------------------------------------------------------------------
 SRID=4326;[POINT Z (1 2 3)@2000-01-01 00:00:00+00, POINT Z (4 5 6)@2000-01-02 00:00:00+00]
(1 row)

SELECT asEWKT(fromEWKB(asEWKB(tgeompoint 'SRID=4326;{[Point(1 2 3)@2000-01-01, Point(4 5 6)@2000-01-02],[Point(1 2 3)@2000-01-03, Point(4 5 6)@2000-01-04]}')));
                                                                                     asewkt                                                                                     
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 SRID=4326;{[POINT Z (1 2 3)@2000-01-01 00:00:00+00, POINT Z (4 5 6)@2000-01-02 00:00:00+00], [POINT Z (1 2 3)@2000-01-03 00:00:00+00, POINT Z (4 5 6)@2000-01-04 00:00:00+00]}
(1 row)

